packages/aws-cdk-lib/aws-sns/lib/subscription.ts

```diff
@@ -136,6 +144,43 @@ export class Subscription extends Resource {
       throw new Error('Subscription role arn is required field for subscriptions with a firehose protocol.');
     }
 
+    if (props.deliveryPolicy) {
+      if (props.protocol !== SubscriptionProtocol.HTTP && props.protocol !== SubscriptionProtocol.HTTPS) {
+        throw new Error('Delivery policy is only supported for HTTP and HTTPS subscriptions');
+      }
+      const { healthyRetryPolicy, throttlePolicy } = props.deliveryPolicy;
+      if (healthyRetryPolicy) {
+        const delayTargetLimit = 3600;
+        if (healthyRetryPolicy.minDelayTarget < 1 || healthyRetryPolicy.minDelayTarget > delayTargetLimit) {
+          throw new Error(`minDelayTarget must be between 1 and ${delayTargetLimit} inclusive`);
+        }
```

Comment:
In order to support Duration, please make the following modifications.

```suggestion
        const minDelayTarget = healthyRetryPolicy.minDelayTarget
        const maxDelayTarget = healthyRetryPolicy.maxDelayTarget
        if (minDelayTarget.toMilliseconds() < Duration.seconds(1).toMilliseconds()) {
          throw new Error(`minDelayTarget must be at least 1 second, got ${minDelayTarget.toMilliseconds()}ms.`);
        }
        if (maxDelayTarget.toMilliseconds() < Duration.seconds(1).toMilliseconds()) {
          throw new Error(`maxDelayTarget must be at least 1 second, got ${maxDelayTarget.toMilliseconds()}ms.`);
        }
        if (minDelayTarget.toSeconds() < 1 || minDelayTarget.toSeconds() > delayTargetLimit) {
          throw new Error(`minDelayTarget must be between 1 and ${delayTargetLimit} seconds inclusive`);
        }
```